### ISSUE TITLE
Consider adding support for third and/or fourth LED and Wii remote

### ISSUE THREAD

**Issue by hifi** [samuelballantyne/IR-Light-Gun#1]
The way the gun is currently calibrated is pretty much spot on with the original Wii remote and naturally should suffer from the same inaccuracy issues when moving around.

If you add third (optional fourth) LEDs you could skip manual calibration almost completely and get true absolute position the gun is pointing at.

Furthermore if you'd be interested to expand this project into supporting a Wii remote it could serve as a generic wired or wireless IR lightgun framework if you have more time and interest than me. The wired real guncon version naturally has less latency but also requires more work.

My somewhat incomplete Wii remote project with test videos: https://github.com/hifi/xwiigun

**Comment by samuelballantyne** [comment]
Someone on GitHub actually pointed me towards your project it looks great. The only problem I thought you might have with it is the distance you would have to stand back from the screen? I feel like I'm nearly at the limit with just 2 LED's or is that more to do with camera I'm using?

**Comment by hifi** [comment]
Indeed, my code is designed to keep tracking the rest of the LEDs if one or two go out-of-sight. Also in reverse direction when only two is seen it guesses which two are visible from the four.

The distance isn't that bad in my opinion because the minimum distance is when all four are visible to the camera when you point directly at the screen, after that it should keep tracking them quite well - minus the bugs in the code.

If you happen to have a Wii remote you could try it out yourself and decide if the concept is worth bringing in to create a combined project. The quality if the proof-of-concept code I wrote isn't high and prone to bugs with the LED orientation detection so it's not necessarily usable as-is but the technique should work.

In my opinion a single project that support different LED and peripheral combinations would be excellent. It's way cheaper and easier to rig the Wii remote one together if you can live with the latency caveat so it could be an entry level option even with only two LEDs or a Wii sensor bar. The automatic four LED configuration has excellent accuracy compared to manually configured two LEDs, though.

**Comment by bobbydilley** [comment]
I'd be interested to see how you both tackle this. I beleive it to be mathematically impossible to know for certain which LEDs are visible out of the 4, some sort of real time tracking algorihtm would have to be used.

In real arcade machines that use this technology, the LEDs are flashed on and off in turn so that the gun can read each one and know exactly which corner it is seeing.
